engine/src/main/java/org/camunda/bpm/dmn/engine/impl/evaluation/DecisionTableEvaluationHandler.java

```diff
@@ -164,12 +163,16 @@ protected DmnEvaluatedDecisionRule evaluateMatchingRule(List<DmnDecisionTableOut
 
   protected VariableContext getLocalVariableContext(DmnDecisionTableInputImpl input, DmnEvaluatedInput evaluatedInput, VariableContext variableContext) {
     if (isNonEmptyExpression(input.getExpression())) {
+      String inputVariableName = evaluatedInput.getInputVariable();
+
       return CompositeVariableContext.compose(
-        SingleVariableContext.singleVariable(evaluatedInput.getInputVariable(), evaluatedInput.getValue()),
+        Variables.createVariables()
+            .putValueTyped(inputVariableName, evaluatedInput.getValue())
+            .putValue("inputVariableName", inputVariableName)
```

Comment:
I would change the ordering of the putValue calls. So if for whatever reason an user decides to call its variable `inputVariableName` it will not be overridden by the second putValue. And would still work with the default implementations. I would also add a test for that case.

```java
Variables.createVariables()
  .putValue("inputVariableName", inputVariableName)
  .putValueTyped(inputVariableName, evaluatedInput.getValue())
```